pulseapi/entries/views.py

```diff
@@ -4,13 +4,48 @@
 
 import django_filters
 from rest_framework import (filters, status)
-from rest_framework.decorators import detail_route
-from rest_framework.generics import ListCreateAPIView, RetrieveAPIView
+from rest_framework.decorators import detail_route, api_view
+from rest_framework.generics import ListCreateAPIView, RetrieveAPIView, ListAPIView
 from rest_framework.pagination import PageNumberPagination
 from rest_framework.response import Response
 
 from pulseapi.entries.models import Entry
 from pulseapi.entries.serializers import EntrySerializer
+from pulseapi.users.models import UserBookmarks
+
+@api_view(['PUT'])
+def toggle_bookmark(request, entryid):
```

Comment:
That makes sense. There's then a little bit of manual state management between the front and back end, but it seems reasonable.